### PR TITLE
Audio Files & Data: Fix File/Data Splits and Compiler Flags

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

This does ~~two~~ three things:

1. Splits the audio files correctly. The three splits that were fixed were 1) the voice recognition unit files (`code_801A4EB0.c` and `code_801A51F0.c`), and 2) the jpegutils functions that were in the audio file `code_801A7B10.c`. This second split is already taken care of in this PR: https://github.com/zeldaret/mm/pull/322, but I want to start making more audio PRs soon, so this does the split separately. EDIT: A third split included is the audio libultra_code files `code_80194710.c` and `code_80194790.c`
2. The audio files are not compiled with the same flags as the rest of code. They do not compile with the `-g3` flag. So I made a directory with all the audio files and fixed the MakeFile to use the correct flags
3. The data files `code_801D2E80.data.s`, `code_801D3D90.data.s`, and `code_801D55B0.data.s` has been properly split into the correct files:
- audio_data.data.s (data file that’s compiled separately at the start of all the audio files)
- audio_synthesis.data.s
- audio_load.data.s
- code_80192BE0.data.s
- code_80194790.data.s
- audio_effects.data.s
- audio_seqplayer.data.s
- code_8019AF00.data.s
- code_801A4EB0.data.s

Edit 2:
More audio splits were made that were missed previously:

The rodata `D_801E1100` from `code_801AA020.c`/`jpegdecoder.c` is actually `audio_init_params.c` from OoT, so that was properly separated into it's own file

`code_801D9090.data.s` was properly split into the proper audio files:
- `audio_sound_params.data.s`
- `code_801A5BD0.data.s`
- `code_801A7B10.data.s`

### CODE REVIEW

**Comment by engineer124** [comment]
Converting to draft because I was going to do another PR splitting audio data correctly, but it's so integrated with this PR that I want to include that here as well

**Comment by engineer124** [comment]
Okay, it's open again. The data files `code_801D2E80.data.s`, `code_801D3D90.data.s`, and `code_801D55B0.data.s` have been properly split into the correct files:
- audio_data.data.s
- audio_synthesis.data.s
- audio_load.data.s
- code_80192BE0.data.s
- code_80194790.data.s
- audio_effects.data.s
- audio_seqplayer.data.s
- code_8019AF00.data.s
- code_801A4EB0.data.s

**Review by AngheloAlf — COMMENTED** [review]
New filesplits are always welcomed!
Just some minor details I noted

File: tools/disasm/variables.txt [hunk 1]
```diff
@@ -2274,11 +2274,9 @@
...
     0x801D4D98:("D_801D4D98","UNK_PTR","",0x4),
     0x801D4DB0:("D_801D4DB0","UNK_PTR","",0x4),
-    0x801D4DB4:("D_801D4DB4","UNK_TYPE1","",0x1),
-    0x801D4FB4:("D_801D4FB4","UNK_TYPE1","",0x1),
-    0x801D51B4:("D_801D51B4","UNK_TYPE4","",0x4),
-    0x801D53B4:("D_801D53B4","UNK_TYPE4","",0x4),
-    0x801D55B4:("D_801D55B4","UNK_TYPE1","",0x1),
+    0x801D4DB4:("D_801D4DB4","f32","[256]",0x4),
+    0x801D51B4:("D_801D51B4","f32","[256]",0x4),
+    0x801D55B4:("D_801D55B4","f32","[128]",0x4),
```
[the remark this answers is not in the document]
Comment (engineer124) [reply tools/disasm/variables.txt:2279]: Ah, you're totally right

**Review by AngheloAlf — APPROVED** [review]

**Review by isghj5 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -172,6 +174,7 @@ build/src/boot_O2_g3/%.o: CC := python3 tools/asm-processor/build.py $(CC) -- $(
 build/src/libultra/%.o: CC := python3 tools/asm-processor/build.py $(CC_OLD) -- $(AS) $(ASFLAGS) --
 
 build/src/code/%.o: CC := python3 tools/asm-processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
+build/src/code/audio%.o: CC := python3 tools/asm-processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
```
Comment (Thar0) [inline Makefile:177]: I suspect the `/` was unintentionally left out?
```suggestion
build/src/code/audio/%.o: CC := python3 tools/asm-processor/build.py $(CC) -- $(AS) $(ASFLAGS) --
```

**Review by Thar0 — COMMENTED** [review]
One thing

**Comment by engineer124** [comment]
More audio splits were made that were missed previously:

The rodata `D_801E1100` from `code_801AA020.c`/`jpegdecoder.c` is actually `audio_init_params.c` from OoT, so that was properly separated into it's own file

`code_801D9090.data.s` was properly split into the proper audio files:
- `audio_sound_params.data.s`
- `code_801A5BD0.data.s`
- `code_801A7B10.data.s`

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**